doc/zmq.md

```diff
@@ -0,0 +1,145 @@
+# Block and Transaction Broadcasting With ZeroMQ
+
+[ZeroMQ](http://zeromq.org/) is a lightweight wrapper around TCP
+connections, inter-process communications, and shared-memory,
+providing various message-oriented semantics such as publish/subcribe,
+request/reply, and push/pull.
+
+The Bitcoin Core daemon can be configured to act as a trusted "border
+router", implementing the bitcoin wire protocol and relay, making
+consensus decisions, maintaining the local blockchain database,
+broadcasting locally generated transactions into the network, and
+providing a queryable RPC interface to interact on a polled basis for
+requesting blockchain related data.  However, there exists only a
+limited service to notify external software of events like the arrival
+of new blocks or transactions.
+
+The ZeroMQ facility binds a "publish" port that broadcasts all newly
+arrived *and validated* blocks and transactions to one or more
+connected subscribers.  This read-only facility requires only the
+connection of a corresponding ZeroMQ subscriber port in receiving
+software; it is not authenticated nor is there any two-way protocol
+involvement.
+
+In this fashion, external software can use a trusted Bitcoin Core
+daemon to do the heavy lifting of communicating with the P2P network,
+while still receiving network information asynchronously in real
+time. As transactions and blocks arrive via the P2P network, Bitcoin
+Core will apply all the validation and standardization rules to this
+data, only passing along through ZeroMQ those items that pass.
+
+ZeroMQ sockets are self-connecting and self-healing; that is, connects
+made between two endpoints will be automatically restored after an
+outage, and either end may be freely started or stopped in any order.
+
+Because ZeroMQ is message oriented, subscribers receive transactions
+and blocks all-at-once and do not need to implement any sort of
+buffering or reassembly.
+
+## Prerequisites
+
+The ZeroMQ feature in Bitcoin Core uses only a very small part of the
+ZeroMQ C API, and is thus compatible with any version of ZeroMQ
+from 2.1 onward, including all versions in the 3.x and 4.x release
+series.  Typically, it is packaged by distributions as something like
+*libzmq-dev*.
+
+The C++ wrapper for ZeroMQ is *not* needed.
+
+## Enabling
+
+By default, the ZeroMQ port functionality is disabled.  Two steps are
+required to enable--compiling in the ZeroMQ code, and configuring
+runtime operation on the command-line or configuration file.
+
+    $ ./configure --enable-zmq (other options)
+
+This will produce a binary that is capable of providing the ZeroMQ
+facility, but will not do so until also configured properly.
+
+## Configuration
+
+Currently, the ZeroMQ facility only needs to have the ZeroMQ endpoint
+specified:
+
+    $ bitcoind -zmqpub=tcp://127.0.0.1/28332
```

Comment:
this should be: `bitcoind -zmqpub=tcp://127.0.0.1:28332` (dash instead slash before port)